catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/CqlCommands.java

```diff
@@ -0,0 +1,225 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+
+package org.codice.ddf.commands.catalog;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Arrays;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.GregorianCalendar;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.karaf.shell.api.action.Option;
+import org.geotools.filter.text.cql2.CQL;
+import org.geotools.filter.text.cql2.CQLException;
+import org.opengis.filter.Filter;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.types.Core;
+import ddf.catalog.filter.FilterBuilder;
+
+public abstract class CqlCommands extends CatalogCommands {
+
+    public static final String DATE_FORMAT = "MM-dd-yyyy";
+
+    public static final String DEFAULT_TEMPORAL_PROPERTY = Core.CREATED;
+
+    private static final String MUTUALLY_EXCLUSIVE_OPTION_MESSAGE =
+            " NOTE: Does not apply to CQL filters. Does not stack with other --lastXXXX options.";
+
+    private static final String LASTXXXX_MESSAGE = MUTUALLY_EXCLUSIVE_OPTION_MESSAGE
+            + " Smaller --lastXXXX time units take precedence over larger time units.";
+
+    private static final String[] TEMPORAL_PROPERTIES =
+            {Core.CREATED, Metacard.EFFECTIVE, Core.EXPIRATION, Core.MODIFIED};
```

Comment:
We could do it dynamically. Pulling in the reference-list of metacard types and looping through the attribute descriptors for any with the `DATE_TYPE`. That would pull any Date types as they change.

In blueprint it would just be a `<reference-list interface="ddf.catalog.data.MetacardType" />` but since karaf commands use the annotations not sure if you can just use `@Reference` or if they have some sort of `reference-list` annotation. Then iterating through that list and calling `MetacardType::getAttributeDescriptors` and pulling out any with the date type should get us all the system date types dynamically.